### ISSUE TITLE
"git source is not yet checked out" error when "bundle" is run from a script

### ISSUE THREAD

**Issue by emirkin** [bundler/bundler#4364]
I modify _Gemfile_, delete _Gemfile.lock_ then run `bundle`.

Next, I get the standard "git source .../my_gem.git" is not yet checked out. Please run bundle install" error.

I do *not* run those commands from bash manually. Instead, I do `rm Gemfile.lock`, followed by `bash --login -c "bundle"` (part of a scripted operation). It is the second command that produces this error.

Strangely, if I run `rm Gemfile.lock; bundle` myself, it works fine. So, it appears that my running `bundle` directly is different from `bash --login -c "bundle"`. What could I be missing?

**Comment by indirect** [comment]
Lots of things could be different—environment variables, working directory, home directory, tons of stuff. Try running `bundle env` and `bundle config` both as yourself and as part of the script and see if there are any differences.

Whatever the problem is, this doesn't sound like a bug in bundler, so I'm going to close this ticket. Good luck figuring it out!

**Comment by emirkin** [comment]
@indirect Thanks for the tip. Those 2 commands throw the same error (from inside the script).

**Comment by indirect** [comment]
Then something even weirder is happening, because neither `bundle config` nor `bundle env` care whether git repos are checked out or not.

**Comment by emirkin** [comment]
@indirect I was able to narrow it down to 1 gem in my gemfile. I changed the owner for that repo from one value to another (both work, but require to update the gem's source in Gemfile.lock).

So, another way to restate my problem is that I _cannot execute any_ `bundle ...` _command_ as long as there is a "pending" change in Gemfile. Even commands such as `bundle config` that are not supposed to look inside my `Gemfile` and `Gemfile.lock` do not work. *That is a bug* in my opinion.

When I fix `Gemfile` to use the original value (so `Gemfile.lock` does not need to change), I get the following for `bundle config`:

_when things work_ (directly from my shell):
```
Set for the current user (/Users/eampro/.bundle/config): "--use-system-libraries"
```
_when things *don't* work (executing command via `bash --login -c "..."`:
```
Set for the current user (/Users/eampro/.bundle/config): "--use-system-libraries"

gemfile
Set via BUNDLE_GEMFILE: "/Users/eampro/projects/XXX/ABC/Gemfile"

bin_path
Set via BUNDLE_BIN_PATH: "/Users/eampro/.rvm/gems/ruby-2.2.3/gems/bundler-1.11.2/exe/bundle"
```

To summarize, there are 2 issues. One is definitely a bug -- `bundle config` requires dependency that it does not really need. Another is my problem with executing a `bundle update` command, which may or may not be an issue (I do not understand why it cannot check out an updated gem location when I run `bundle update` from within a non-interactive shell).

**Comment by emirkin** [comment]
To follow up, it appears that my problem would go away if I could make `bundle update` ignore the additional BUNDLE_GEMFILE and BUNDLE_BIN_PATH vars it gets when I run the command from a spun off shell process. Is there a way to do that?

**Comment by emirkin** [comment]
http://stackoverflow.com/questions/10185552/bundle-install-runs-from-incorrect-directory solved my problem.

So, the only bug that remains is that innocuous commands such as `bundle config` depend on the state of my Gemfile.

**Comment by indirect** [comment]
@emirkin I can't reproduce your issue, yet. Please provide exact reproduction steps. Here's an example of me trying and failing to reproduce your issue:

```
💙   andre.arko ~/src/bundler/testcases/4364  bundle config
Settings are listed in order of priority. The top value will be used.
jobs
Set via BUNDLE_JOBS: "8"

```
$ ls
$ echo "gem 'foo'" >> Gemfile
$ bundle config
Settings are listed in order of priority. The top value will be used.
jobs
Set via BUNDLE_JOBS: "8"

$ bundle show
Could not find gem 'foo (>= 0) ruby' in the gems available on this machine.
```
